src/main/kotlin/org/opensearch/replication/seqno/RemoteClusterRetentionLeaseHelper.kt

```diff
@@ -60,15 +73,60 @@ class RemoteClusterRetentionLeaseHelper constructor(val followerClusterName: Str
             return true
         }
         catch (e: RetentionLeaseNotFoundException) {
+            return AddNewRetentionLeaseIfOldExists(leaderShardId, followerShardId, seqNo)
```

Comment:
Could you explain why we are adding new retention lease after verifying that old one doesn't exist?

Reply:
We are always checking for new retention lease first as the default.
So if new is not found, we check for old and add new if old exists